Build error with clang 6 on linux and shorthand tuple initialization

> MangeliS: I tried to build with export CXX=/usr/bin/clang++-6.0 &&  cmake .. -DCMAKE_BUILD_TYPE=Debug -DwxWidgets_PREFIX=/usr -DCMAKE_CXX_FLAGS=-std=c++17   && cmake --build . --target TrenchBroom -- -j4
MangeliS: but it fails with [ 30%] Building CXX object CMakeFiles/common.dir/common/src/View/EntityAttributeSelectedCommand.cpp.o
/home/mankeli/src/TrenchBroom/common/src/View/ScaleObjectsToolController.cpp:96:20: error: chosen constructor is explicit in copy-initialization
            return {restricter, snapper, initialPoint};
                   ^~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/tuple:479:19: note: explicit constructor declared here
        constexpr tuple(_UElements&&... __elements)
                  ^
[11:15 PM] MangeliS: (linux, clang 6.0,  latest wxwidgets dev build)

See also https://stackoverflow.com/a/16713413 why this type tuple initialization is an error.

Same info from Discord with some additions that I originally posted to an another thread.
```
export CXX=/usr/bin/clang++-6.0
export CC=/usr/bin/clang-6.0
cmake .. -DCMAKE_BUILD_TYPE=Debug -DwxWidgets_PREFIX=/usr -DCMAKE_CXX_FLAGS=-std=c++17
cmake --build . --target TrenchBroom -- -j4
```
And it fails with
```
[ 30%] Building CXX object CMakeFiles/common.dir/common/src/View/EntityAttributeSelectedCommand.cpp.o
/home/mankeli/src/TrenchBroom/common/src/View/ScaleObjectsToolController.cpp:96:20: error: chosen constructor is explicit in copy-initialization
            return {restricter, snapper, initialPoint};
                   ^~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/tuple:479:19: note: explicit constructor declared here
        constexpr tuple(_UElements&&... __elements)
                  ^
```

during configure I also get messages like:
```
CMake Error in CMakeLists.txt:
  Target "TrenchBroom-Test" requires the language dialect "CXX17" (with
  compiler extensions), but CMake does not know the compile flags to use to
  enable it.
```

(ubuntu, clang 6.0)

--- a/common/src/View/ScaleObjectsToolController.cpp
+++ b/common/src/View/ScaleObjectsToolController.cpp
@@ -93,7 +93,7 @@ namespace TrenchBroom {
                 return p;
             }();
 
-            return {restricter, snapper, initialPoint};
+            return std::make_tuple(restricter, snapper, initialPoint);
         }
 
         static std::pair<AnchorPos, ProportionalAxes> modifierSettingsForInputState(const InputState& inputState) {

--- a/common/src/View/ShearObjectsToolController.cpp
+++ b/common/src/View/ShearObjectsToolController.cpp
@@ -86,7 +86,7 @@ namespace TrenchBroom {
                 snapper = new LineDragSnapper(grid, sideways);
             }
 
-            return {restricter, snapper};
+            return std::make_tuple(restricter, snapper);
         }
 
         void ShearObjectsToolController::doModifierKeyChange(const InputState& inputState) {

